### ISSUE TITLE
Question: is it possible to build against libopus v1.5.(2) to be able to use the new features #108

### ISSUE THREAD

**Issue by wojciak** [hraban/opus#54]
https://opus-codec.org/release/stable/2024/03/04/libopus-1_5.html

**Comment by hraban** [comment]
I'm curious too 👍  Give it a shot and let me know!

**Comment by avezina-ubik** [comment]
Hey @hraban,

Tried compiling it quickly on my machine and it seems to be working fine. I simply added `SetComplexity` to the decoder and everything compiles and runs smoothly (more tests needed of course).

Before going further, do you have an idea on how we could handle libopus>=1.5 diiferently than prior versions ?

1. Go module versions
1. Build tags
1. Lock the library to a specific version of libopus

If it were me I would go with module version, but let me know what you think.

**Comment by hraban** [comment]
Afaik this is all dynamically linked so the client can just choose what version to link? It's been a while since I've done anything with this but last I checked the build and run system weren't linked like that. Does this not work for your purposes?

**Comment by avezina-ubik** [comment]
You are right, it seems to compile anyway since the "new feature" is actually passing through `opus_decoder_ctl`

I will submit a PR shortly to add the SetComplexity to the decoder

**Comment by avezina-ubik** [comment]
This should help #54
